fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -2444,13 +2447,32 @@ private Bundle createSearchBundle(List<Resource> resources, FHIRSearchContext se
                                             .id(UUID.randomUUID().toString())
                                             .total(totalCount);
 
+        // Calculate how many resources are 'match' mode
+        int pageSize = searchContext.getPageSize();
+        int offset = (searchContext.getPageNumber() - 1) * pageSize;
+        int matchResourceCount = pageSize;
+        if (totalCount.getValue() < offset + pageSize) {
+            matchResourceCount = totalCount.getValue() - offset;
+        }
+
+        // Check if too many included resources
+        if (resources.size() > matchResourceCount + SearchConstants.MAX_PAGE_SIZE) {
+            String msg = "Number of returned 'include' resources exceeds allowable limit of " + SearchConstants.MAX_PAGE_SIZE;
+            throw buildRestException(msg, IssueType.BUSINESS_RULE, IssueSeverity.ERROR);
+        }
+
         for (Resource resource : resources) {
             if (resource.getId() == null) {
                 throw new IllegalStateException("Returned resources must have an id.");
             }
-            Bundle.Entry entry = Bundle.Entry.builder().fullUrl(Uri.of(getRequestBaseUri(type) + "/"
-                    + resource.getClass().getSimpleName() + "/"
-                    + resource.getId())).resource(resource).build();
+            Bundle.Entry entry = Bundle.Entry.builder()
+                    .fullUrl(Uri.of(getRequestBaseUri(type) + "/" + resource.getClass().getSimpleName() + "/" + resource.getId()))
+                    .resource(resource)
+                    .search(Search.builder()
+                        .mode(matchResourceCount-- > 0 ? SearchEntryMode.MATCH : SearchEntryMode.INCLUDE)
+                        .score(Decimal.of("1"))
+                        .build())
+                    .build();
 
             bundleBuider.entry(entry);
```

Comment:
not your typo, but if you happen to make another round of edits, please fix this variable name bundleBuider -> bundleBuilder

Reply:
Fixed.